src/components/Profile.js

```diff
@@ -1,24 +1,31 @@
 import React, { Component } from 'react';
+import { Link } from "react-router-dom";
 import './Profile.css';
 
 class Profile extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      // name: props.location.state.user.name,
-      // favorites: props.location.state.user.favorites
+      name: this.props.user ? this.props.user.name : '',
+      favorites: this.props.user ? this.props.user.favorites : []
     }
-    console.log(this.props)
   }
 
   render() {
     const {name, favorites} = this.state;
 
+    const favoriteDisplay = favorites.map(movie => {
+      return <Link className='link-to-favorite' to={{
+          pathname: `/movie-details/${movie.title.replace(/\s+/g, '')}`,
+          state: {movie: movie, user: this.props.user}
+        }}><h3 key={movie.id}>{movie.title}</h3></Link>
+    })
+
```

Comment:
Love this! Will possibly have a merge conflict here when i push my work up because i had dabbled with this as well but I'm sure we can solve it relatively easily